commitment/commitment_test.go

```diff
@@ -1177,3 +1184,52 @@ func TestIsTaprootAssetCommitmentScript(t *testing.T) {
 	require.True(t, IsTaprootAssetCommitmentScript(testTapCommitmentScript))
 	require.False(t, IsTaprootAssetCommitmentScript(TaprootAssetsMarker[:]))
 }
+
+// TestAssetCommitmentNoWitness tests that an asset commitment of a v1 asset is
+// the same with and without the witness field of the asset set.
+func TestAssetCommitmentNoWitness(t *testing.T) {
+	t.Parallel()
+
+	// We'll start by generating a random asset.
+	genesis1 := asset.RandGenesis(t, asset.Normal)
+	groupKey1 := asset.RandGroupKey(t, genesis1)
+	asset1 := randAsset(t, genesis1, groupKey1)
+
+	// We'll modify the asset version so it uses the segwit encoding for
+	// the leaf commitment.
+	asset1.Version = asset.V1
+
+	// We'll modify the asset witness to ontain a valid prev ID, along with
+	// some random data.
+	var randTxid chainhash.Hash
+	_, err := rand.Read(randTxid[:])
+	require.NoError(t, err)
+
+	asset1.PrevWitnesses[0].PrevID.OutPoint = wire.OutPoint{
+		Index: uint32(rand.Int()),
+		Hash:  randTxid,
+	}
+	asset1.PrevWitnesses[0].PrevID.ID = asset.ID(randTxid)
+	copy(asset1.PrevWitnesses[0].PrevID.ScriptKey[:], randTxid[:])
+
+	// Now that we have all our initila information set, we'll make a copy
+	// of the above asset. One of them will have a witness vector field,
+	// the other won't.
+	assetNoWitness := asset1.Copy()
+
+	asset1.PrevWitnesses[0].TxWitness = [][]byte{randTxid[:]}
+
+	// Next, we'll use the assets to create two root tap commitments.
+	commitmentWitness, err := FromAssets(asset1)
+	require.NoError(t, err)
+
+	commitmentNoWitness, err := FromAssets(assetNoWitness)
+	require.NoError(t, err)
+
+	// The two commitment should be identical as this asset version leaves
+	// out the tx witness field in the leaf encoding.
+	require.Equal(
```

Comment:
Should we also assert that the same asset with `V0` has a _different_ commitment? Probably kind of pointless since changing the version itself already changes the leaf... So probably not.

Reply:
Why not, tacked that on!